cortex-r-rt/src/lib.rs

```diff
@@ -487,6 +483,8 @@ core::arch::global_asm!(
     "#,
     restore_context!(),
     r#"
+        // overwrite the saved LR with the adjusted one
```

Comment:
Am I understanding it correctly that the LR returned by the user is stored here, which does not necessarily have to be the adjusted LR?

Reply:
yeah, fair, the handler *could have* but *might not have* adjusted the value

Reply:
fixed